webapp/channels/src/components/edit_post/edit_post.tsx

```diff
@@ -237,13 +237,15 @@ const EditPost = ({editingPost, actions, canEditPost, config, channelId, draft,
         }
 
         let updatedPost = {
-            ...editingPost.post,
             message: editText,
+            id: editingPost.postId,
+            channel_id: editingPost.post.channel_id,
         };
 
         const hookResult = await actions.runMessageWillBeUpdatedHooks(updatedPost, editingPost.post);
         if (hookResult.error) {
             setPostError(hookResult.error);
```

Comment:
I'm wondering if we need to do the same `setErrorClass` stuff we're doing below. I'm not too familiar with the control flow of this file/component. What is the visual behavior that occurs when the plugin returns an error?

Reply:
Yes, as things stand we need to fix this because `setPostError` now expects a `React.ReactNode`. So at a minimum we should do something like:

```js
setPostError((<>{hookResult.error.message}</>));
```

That would also allow the plugin to potentially pass a node directly so that they get more control on formatting.

Reply:
@streamer45 `React.ReactNode` is purposely permissive - essentially anything "renderable", including strings and numbers. So passing a string works for this, which is why CI is passing on the PR's current state.

> That would also allow the plugin to potentially pass a node directly so that they get more control on formatting.

I'm not sure we want the plugin to be able to have that much freedom on this particular hook. 1/5 I think a string is a fine type to expect from the hook error message.

Reply:
@mickmister I think CI passes because this is likely not tested anywhere. The API as it stands is expecting an object of typer `{error: {message: 'error message'}}` so at a minimum we need to pass `hookResult.error.message` unless we change the contract. 

I think I am coming from the point of view of someone extending MM quite often, and personally I wouldn't mind if we kept things a little more flexible as making breaking changes later is always more painful. But not a big deal.

Reply:
@qryptdev Just checking in to see if next steps are clear? If we want to get this into 8.1 the deadline is Friday EOD. Please let me know if I can assist in place of streamer45 who is off until Monday.

Reply:
We had a bit of internal delay. The code has been updated per suggestion.